### PR TITLE
Allow warp-tls to request client certificates.

### PR DESCRIPTION
**Pull request by clinty**

### CODE REVIEW

**Comment by snoyberg** [comment]
Looks good to me. One request: could you add a little bit of documentation to these new fields explaining what their values mean? Even just pointing at the relevant TLS fields would be a good addition.

**Comment by kazu-yamamoto** [comment]
Looks good to me, too.

**Comment by clinty** [comment]
Sure, how's this?

File: warp-tls/Network/Wai/Handler/WarpTLS.hs [hunk 1]
```diff
@@ -81,6 +83,24 @@ data TLSSettings = TLSSettings {
...
+    -- is set to True, you must handle received certificates in a server hook
+    -- or all connections will fail.
+    --
+    -- Default: False
+  , tlsServerHooks :: TLS.ServerHooks
+    -- ^ Whether or not to demand a certificate from the client.  If this
+    -- is set to True, you must handle received certificates in a server hook
+    -- or all connections will fail.
+    --
+    -- Default: False
```
Comment (MaxGabriel) [inline warp-tls/Network/Wai/Handler/WarpTLS.hs]: Is the `tlsWantClientCert` documentation supposed to be == to the `tlsServerHooks` documentation?

**Comment by clinty** [comment]
@MaxGabriel well, that was embarrassing

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**